src/common/options/global.yaml.in

```diff
@@ -6834,3 +6834,21 @@ options:
   level: dev
   default: 0
   desc: When EC writes should generate PDWs (development only) 0=optimal 1=never 2=when possible
+- name: osd_crush_scaling_factor
```

Comment:
new options should be added to the docs with a ``confval::`` directive, and they'll get the description from here.

the warning could be added to the rados/troubleshooting section too

Reply:
I'm worried about this being a normal config.  Imagine that a user see this config and sets this to 10.0 after the cluster has existed for a while.  They'd probably imagine that it would actually change the existing crush map.  Instead, any newly added OSDs would only get 1/10.0 of the data the user expects.

It would also be very strange for this value to be different on different OSDs or specified on the command line as it would lead to it being different across the cluster.

I think this needs to be a member of the OSDMap itself, or otherwise stored in the monitor.  The OSD probably should specify its *size* in update_crush_location with the mon then generating the crush weight from that size and the single-source-of-truth osd_crush_scaling_factor.

@jdurgin thoughts?

Reply:
agreed, the osdmap approach would be more failsafe